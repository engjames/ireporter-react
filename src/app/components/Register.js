import React, { Component } from "react";
import "../../style/signup.scss";
import '../../style/app.scss';
import { Link } from "react-router-dom";

class Register extends Component {
  render() {
    const {
      handleChange,
      handleSignup
    } = this.props;
    return (
      <React.Fragment>
        <div className="row">
          <div className="col-md-4" />
          <div className="col-md-4">
            <div className="card">
              <form id="signup-form"
                onSubmit={handleSignup}>
                <input
                  onChange={handleChange}
                  type="text"
                  id="lastname"
                  placeholder="lastname"
                  className="form-control input-field"
                />
                <br/>
                <input
                  onChange={handleChange}
                  type="text"
                  id="firstname"
                  placeholder="firstname"
                  className="form-control input-field"
                />
                <br/>
                <input
                  onChange={handleChange}
                  type="text"
                  id="email"
                  placeholder="email"
                  className="form-control input-field"
                />
                <br/>
                <input
                  onChange={handleChange}
                  type="password"
                  id="password"
                  placeholder="password"
                  className="form-control input-field"
                />
                <br/>
                <button className="signup-btn" type="submit">Signup</button>
                <p className="have-an-account">Already have an account!
                  <Link to="/login">Login</Link></p>
              </form>
            </div></div>
          <div className="col-md-4" />
        </div>
      </React.Fragment>
    );
  }
}

export default Register;
