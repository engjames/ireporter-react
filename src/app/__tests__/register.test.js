import React from 'react';
import { shallow , mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import Register from '../../app/components/Register';

describe('<Register />', () => {
  let wrapper;
  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});


