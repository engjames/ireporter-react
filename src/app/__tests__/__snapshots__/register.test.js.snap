// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> matches snapshot 1`] = `
<Fragment>
  <div
    className="row"
  >
    <div
      className="col-md-4"
    />
    <div
      className="col-md-4"
    >
      <div
        className="card"
      >
        <form
          id="signup-form"
        >
          <input
            className="form-control input-field"
            id="lastname"
            placeholder="lastname"
            type="text"
          />
          <input
            className="form-control input-field"
            id="firstname"
            placeholder="firstname"
            type="text"
          />
          <input
            className="form-control input-field"
            id="email"
            placeholder="email"
            type="text"
          />
          <input
            className="form-control input-field"
            id="password"
            placeholder="password"
            type="password"
          />
          <button
            className="signup-btn"
            type="submit"
          >
            Signup
          </button>
          <p
            className="have-an-account"
          >
            Already have an account!
            <Link
              to="/login"
            >
              Login
            </Link>
          </p>
        </form>
      </div>
    </div>
    <div
      className="col-md-4"
    />
  </div>
</Fragment>
`;
